packages/fx-core/src/plugins/resource/sql/sqlClient.ts

```diff
@@ -26,18 +27,22 @@ export class SqlClient {
         return false;
       }
     } catch (error) {
-      this.ctx.logProvider?.error(
-        ErrorMessage.SqlCheckDBUserError.message(this.config.identity, error.message)
-      );
-      if (error?.message?.includes(ErrorMessage.AccessMessage)) {
+      if (error?.message?.includes(ErrorMessage.AccessMessage) && extractIp(error.message)) {
+        const ip = extractIp(error.message);
```

Comment:
better to cache the extracted value in the previous if condition.

Reply:
it is fixed. Thanks.